### PR TITLE
[12.0][MIG] hr_expense_sequence

### PR DESCRIPTION
**Pull request by ps-tubtim**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
There's already other migration in #617. Please review that one.

**Comment by ps-tubtim** [comment]
Thank you @pedrobaeza.

**Comment by kittiu** [comment]
@pedrobaeza, sorry, I closed that one already, as I want our teammate to redo it again from 10->12
@ps-tubtim, may be you can reopen this one and continue?

**Comment by pedrobaeza** [comment]
Ok, reopening. Please check Travis status and squash together adjacent "OCA Transbot.." commits using https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

**Comment by ps-tubtim** [comment]
Ok, I will squash. Thank you.

**Review by AaronHForgeFlow — APPROVED** [review]
LGTM :+1:

**Comment by kittiu** [comment]
@ps-tubtim please add test script to make codecov become green.

File: hr_expense_sequence/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    'depends': [
+        'hr_expense',
+    ],
+    'data': [
+        'data/hr_expense_data.xml',
+        'views/hr_expense_expense_view.xml',
+        'report/report_expense_sheet.xml',
+    ],
+    'installable': True,
+    "post_init_hook": "assign_old_sequences",
```
Comment (kittiu) [inline hr_expense_sequence/__manifest__.py]: @ps-tubtim just a little nicer, if we can use all single quote?

File: hr_expense_sequence/README.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+.. image:: https://odoo-community.org/logo.png
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+This module is part of the `OCA/hr <https://github.com/OCA/hr/tree/12.0-mig-hr_expense_sequence/hr_expense_sequence>`_ project on GitHub.
```
Comment (kittiu) [inline hr_expense_sequence/README.rst]: URL is linked to wrong location, it should be,
https://github.com/OCA/hr/tree/12.0/hr_expense_sequence

Actually, you don't need to create README.rst file. From what I understand, the README.rst will be created by Bot automatically. Just make sure you have the good content in readme folder.

**Comment by kittiu** [comment]
@ps-tubtim please squash commits.
@pedrobaeza , is the ocabot will already helps to squash commits?

**Comment by pedrobaeza** [comment]
If they are properly marked as fixup!, they will, but unfortunately who knows about fixup would probably do git amend instead, so on that cases, (new independent commits) won't be covered.

**Comment by sbidoul** [comment]
Fixups are nice to communicate incremental changes to reviewers (better than a rebase/force push) and the bot will squash them on merge. But indeed fixup commits need to be created very carefully for that to work.

**Comment by pedrobaeza** [comment]
With GitHub there's no need nowadays to do fixups for being able to see incremental changes, so I prefer to directly amend migration commit.

**Comment by kittiu** [comment]
LGTM

File: hr_expense_sequence/tests/test_hr_expense_sequence.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            'unit_amount': self.product.standard_price,
+            'quantity': 1,
+        })
+        expense.action_submit_expenses()
+        return expense
+
+    def test_create_sequence(self):
+        self.sheet.expense_line_ids = [
+            (6, 0, [self.expense.id])]
+        self.sheet.action_submit_sheet()
```
Comment (kittiu) [inline hr_expense_sequence/tests/test_hr_expense_sequence.py]: When doing test, you need to Assert the value to make sure that the result is as expected.
I.e., in this case, you may expect the new number?

**Review by Saran440 — APPROVED** [review]
LGTM :+1:

File: hr_expense_sequence/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'HR expense sequence',
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+    'category': 'Human Resources',
+    'author': "Serv. Tecnol. Avanzados - Pedro M. Baeza,"
+              "Odoo Community Association (OCA)",
+    'website': 'http://www.serviciosbaeza.com',
```
Comment (nikul-serpentcs) [inline hr_expense_sequence/__manifest__.py]: @ps-tubtim URL should be `https://github.com/OCA/hr`

**Review by nikul-serpentcs — APPROVED** [review]
Minor change

Otherwise LGTM :+1:

**Comment by ps-tubtim** [comment]
@nikul-serpentcs Thank you for review.

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]
As @pedrobaeza already mentioned, `OCA Transbot updated translations from Transifex` commits have to be squashed

File: hr_expense_sequence/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# © 2014 Serv. Tecnol. Avanzados (http://www.serviciosbaeza.com)
```
Comment (alexey-pelykh) [inline hr_expense_sequence/__manifest__.py]: `©` should to be replaced with `Copyright`, see sample https://github.com/OCA/maintainer-tools/tree/master/template/module/

File: hr_expense_sequence/__init__.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+from . import models
```
Comment (alexey-pelykh) [inline hr_expense_sequence/__init__.py:3]: Copyright/License would be great

File: hr_expense_sequence/readme/DESCRIPTION.rst [hunk 7]
```diff
@@ -0,0 +1,7 @@
+This module is maintained by the OCA.
```
Comment (alexey-pelykh) [inline hr_expense_sequence/readme/DESCRIPTION.rst]: Please replace this content with a valid description what this module does.

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

File: hr_expense_sequence/__init__.py (hunk 6 above)
Comment (etobella) [reply hr_expense_sequence/__init__.py:3]: I think it is not necessary on init files.

Comment (alexey-pelykh) [reply hr_expense_sequence/__init__.py:3]: Well, copyright - no, license - yes. But this file also contains code - that should've been separate and with copyright.

Comment (etobella) [reply hr_expense_sequence/__init__.py:3]: Well this a small hole :cry:
I have seen PR asking for the removal. I think the best would be to have a global agreement on OCA. However, It is probably enough with the license

Comment (alexey-pelykh) [reply hr_expense_sequence/__init__.py:3]: Yeah, it's just matter of keeping things clean and making them cleaner

**Comment by ps-tubtim** [comment]
@alexey-pelykh @etobella Thank you for review.

**Comment by kittiu** [comment]
@ps-tubtim only PSC Maintainer can merge this PR. 
To use ocabot to merge, you need to be maintainer first.
Note: you can request to be maintainer with another PR.

**Review by jarroyomorales — APPROVED** [review]
LGTM! Just squash the commits and someone will merge it :wink:

**Comment by ps-tubtim** [comment]
@jarroyomorales Thanks, squashed.
@pedrobaeza Would you help merge this?
We would like to add some module to dependency this module.

**Comment by pedrobaeza** [comment]
@alexey-pelykh has some requested changes, so it should be him that unblock the PR.

File: hr_expense_sequence/data/hr_expense_data.xml [hunk 8]
```diff
@@ -1,4 +1,3 @@
-<?xml version="1.0"?>
```
Comment (alexey-pelykh) [inline hr_expense_sequence/data/hr_expense_data.xml]: I really see no point in removing this, even more - probably it should have been `<?xml version="1.0" encoding="UTF-8" ?>`

File: hr_expense_sequence/report/report_expense_sheet.xml [hunk 9]
```diff
@@ -1,10 +1,10 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (alexey-pelykh) [inline hr_expense_sequence/report/report_expense_sheet.xml]: Why this is removed?

File: hr_expense_sequence/views/hr_expense_expense_view.xml [hunk 10]
```diff
@@ -1,4 +1,3 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (alexey-pelykh) [inline hr_expense_sequence/views/hr_expense_expense_view.xml]: Same question, why this removal was needed?

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]
3 more comments to address and I'll approve

File: hr_expense_sequence/views/hr_expense_expense_view.xml (hunk 10 above)
Comment (ps-tubtim) [reply hr_expense_sequence/views/hr_expense_expense_view.xml]: @alexey-pelykh Sorry, I misunderstand. Edied.

File: hr_expense_sequence/report/report_expense_sheet.xml (hunk 9 above)
Comment (ps-tubtim) [reply hr_expense_sequence/report/report_expense_sheet.xml]: @alexey-pelykh Sorry, I misunderstand. Edied.

**Review by alexey-pelykh — APPROVED** [review]

**Pull request merged**